File: backend/src/main/kotlin/messages/incoming/Lobby.kt
package messages.incoming

import dto.CreateLobbyDTO
import dto.IdDTO
import dto.NameDTO
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonObject

const val IN_NAME = "NAME_MESSAGE"

@Serializable
class NameMessage(override val payload: NameDTO) : IncomingMessage(IN_NAME)

const val IN_CREATE_LOBBY = "CREATE_LOBBY"

@Serializable
class CreateLobbyMessage(override val payload: CreateLobbyDTO): IncomingMessage(IN_CREATE_LOBBY)

const val IN_JOIN_LOBBY = "JOIN_LOBBY"

@Serializable
class JoinLobbyMessage(override val payload: IdDTO): IncomingMessage(IN_JOIN_LOBBY)

const val IN_LEAVE_LOBBY = "LEAVE_LOBBY"

@Serializable
class LeaveLobbyMessage(override val payload: JsonObject): IncomingMessage(IN_LEAVE_LOBBY)

const val IN_READY = "READY"

@Serializable
class ReadyMessage(override val payload: JsonObject): IncomingMessage(IN_READY)

File: frontend/src/app/messages/outgoing/lobby.ts
import { OutgoingMessage } from './common'
import {CreateLobbyDTO, IdDTO, NameDTO} from '../../dto/server'

export const OUT_NAME = 'NAME_MESSAGE'

export class NameMessage extends OutgoingMessage {
  constructor(public payload: NameDTO) {
    super(OUT_NAME, payload)
  }
}

export const OUT_CREATE_LOBBY = 'CREATE_LOBBY'

export class CreateLobbyMessage extends OutgoingMessage {
  constructor(public payload: CreateLobbyDTO) {
    super(OUT_CREATE_LOBBY, payload)
  }
}


export const OUT_JOIN_LOBBY = 'JOIN_LOBBY'

export class JoinLobbyMessage extends OutgoingMessage {
  constructor(public payload: IdDTO) {
    super(OUT_JOIN_LOBBY, payload)
  }
}

export const OUT_LEAVE_LOBBY = 'LEAVE_LOBBY'

export class LeaveLobbyMessage extends OutgoingMessage {
  constructor() {
    super(OUT_LEAVE_LOBBY, {})
  }
}
